File: DaS.ScriptLib.CS/Game/Data/Structures/MapEntry.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using DaS.ScriptLib.Injection;

namespace DaS.ScriptLib.Game.Data.Structures
{

	public class MapEntry
	{


		public readonly int Pointer;
		public MapEntry(int ptr)
		{
			Pointer = ptr;
		}

		public int PointerToBlockAndArea {
			get { return Hook.RInt32(Pointer + 0x4); }
		}

		public byte Block {
			get { return Hook.RByte(PointerToBlockAndArea + 0x6); }
		}

		public byte Area {
			get { return Hook.RByte(PointerToBlockAndArea + 0x7); }
		}

		public int EntityCount {
			get { return Hook.RInt32(Pointer + 0x3c); }
		}

		public int StartOfEntityStruct {
			get { return Hook.RInt32(Pointer + 0x40); }
		}

		public EntityHeader[] GetEntityHeaders()
		{
			EntityHeader[] result = new EntityHeader[EntityCount];

			for (i = 0; i <= result.Length - 1; i++) {
				result[i] = new EntityHeader(StartOfEntityStruct + (EntityHeader.Size * i));
			}

			return result;
		}

		public Entity[] GetEntities()
		{
			Entity[] result = new Entity[EntityCount];

			for (i = 0; i <= EntityCount - 1; i++) {
				result[i] = new EntityHeader(StartOfEntityStruct + (EntityHeader.Size * i)).Entity;
			}

			return result;
		}

		public EntityLocation[] GetEntityLocations()
		{
			EntityLocation[] result = new EntityLocation[EntityCount];

			for (i = 0; i <= EntityCount - 1; i++) {
				result[i] = new EntityHeader(StartOfEntityStruct + (EntityHeader.Size * i)).Location;
			}

			return result;
		}

	}

}

File: DaS.ScriptLib.CS/Game/Data/Structures/MapEntry.vb
Imports DaS.ScriptLib.Injection

Namespace Game.Data.Structures

    Public Class MapEntry

        Public ReadOnly Pointer As Integer

        Public Sub New(ptr As Integer)
            Pointer = ptr
        End Sub

        Public ReadOnly Property PointerToBlockAndArea As Integer
            Get
                Return RInt32(Pointer + &H4)
            End Get
        End Property

        Public ReadOnly Property Block As Byte
            Get
                Return RByte(PointerToBlockAndArea + &H6)
            End Get
        End Property

        Public ReadOnly Property Area As Byte
            Get
                Return RByte(PointerToBlockAndArea + &H7)
            End Get
        End Property

        Public ReadOnly Property EntityCount As Integer
            Get
                Return RInt32(Pointer + &H3C)
            End Get
        End Property

        Public ReadOnly Property StartOfEntityStruct As Integer
            Get
                Return RInt32(Pointer + &H40)
            End Get
        End Property

        Public Function GetEntityHeaders() As EntityHeader()
            Dim result(EntityCount - 1) As EntityHeader

            For i = 0 To result.Length - 1
                result(i) = New EntityHeader(StartOfEntityStruct + (EntityHeader.Size * i))
            Next

            Return result
        End Function

        Public Function GetEntities() As Entity()
            Dim result(EntityCount - 1) As Entity

            For i = 0 To EntityCount - 1
                result(i) = New EntityHeader(StartOfEntityStruct + (EntityHeader.Size * i)).Entity
            Next

            Return result
        End Function

        Public Function GetEntityLocations() As EntityLocation()
            Dim result(EntityCount - 1) As EntityLocation

            For i = 0 To EntityCount - 1
                result(i) = New EntityHeader(StartOfEntityStruct + (EntityHeader.Size * i)).Location
            Next

            Return result
        End Function

    End Class

End Namespace
